### PR TITLE
Added support for all .NET targets

### PR DESCRIPTION
**Pull request by dahall**
I couldn't get @johnsimons repository to give me access, so I took his changes, added my corrections and created a new PR. If you accept this, it will mean you do not need to accept the PR from @johnsimons.

### CODE REVIEW

**Comment by johnsimons** [comment]
@dahall it looks like appveyor still not happy 😞

**Comment by dahall** [comment]
@johnsimons All fixed. It needed to restore NuGet packages before build.

**Comment by johnsimons** [comment]
That is awesome 👏

**Comment by dahall** [comment]
@ricardopolo I put some time into this and have it building and tested. I've also cleaned up the code and added component attributes. I'd appreciate it if you could merge and push to NuGet. There are a few open issues that can be closed as well (#29, #28, and #16).

File: Tulpep.ActiveDirectoryObjectPicker/ExtensionAttributeFor.NET_2.0.cs [hunk 1]
```diff
@@ -1,8 +1,6 @@
 namespace System.Runtime.CompilerServices
 {
-	/// <summary>
-	/// Attribute allowing extenders to be used with .NET Framework 2.0.
-	/// </summary>
+	/// <summary>Attribute allowing extenders to be used with .NET Framework 2.0.</summary>
 	internal sealed class ExtensionAttribute : Attribute
```
[the remark this answers is not in the document]
Comment (dahall) [reply Tulpep.ActiveDirectoryObjectPicker/ExtensionAttributeFor.NET_2.0.cs]: Good catch. I've added it and will commit.

File: Tulpep.ActiveDirectoryObjectPicker/UnmanagedArrayOfStrings.cs [hunk 2]
```diff
@@ -4,49 +4,48 @@
...
-                }
-            }
-        }
+		/// <summary>Initializes a new instance of the <see cref="UnmanagedArrayOfStrings"/> class.</summary>
+		/// <param name="strings">The strings to pack.</param>
+		public UnmanagedArrayOfStrings(IList<string> strings)
+		{
+			if (strings != null)
+			{
+				var _length = strings.Count;
```
[the remark this answers is not in the document]
Comment (dahall) [reply Tulpep.ActiveDirectoryObjectPicker/UnmanagedArrayOfStrings.cs]: Easy fix to add consistency. Done.

File: Tulpep.ActiveDirectoryObjectPicker/Tulpep.ActiveDirectoryObjectPicker.csproj [hunk 3]
```diff
@@ -1,197 +1,41 @@
...
+    <Reference Include="System" />
+    <Reference Include="System.Drawing" />
+    <Reference Include="System.Windows.Forms" />
+    <Reference Include="System.Design" />
   </ItemGroup>
   <ItemGroup>
     <None Include="Tulpep.ActiveDirectoryObjectPicker.snk" />
   </ItemGroup>
-  <ItemGroup>
-    <COMReference Include="ActiveDs">
```
[the remark this answers is not in the document]
Comment (dahall) [reply Tulpep.ActiveDirectoryObjectPicker/Tulpep.ActiveDirectoryObjectPicker.csproj]: This unnecessary. The COM libraries are loaded via their CLSID with interfaces declared directly in the code (see Tulpep.ActiveDirectoryObjectPicker/ComInterop.cs).

**Comment by ricardopolo** [comment]
Thank you so much @dahall, @johnsimons and @MarcelGosselin. This is a fantastic PR. 👏 👏 
Everything works and we have CI working as expected. I released a new nuget version, and bump so we start now from the version 3.0!

Best regard!!

**Pull request merged**